### PR TITLE
Fixes #29285 - Deprecate Rackspace Compute Resource

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: config/initializers/deprecations.rb [hunk 1]
```diff
@@ -1 +1,28 @@
 ActiveSupport::Deprecation.behavior = :silence if Rails.env.production?
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    next if Foreman.in_rake?
+    next unless Rails.env.production?
+    next unless ComputeResource.all.unscoped.map { |cr| cr.class == Foreman::Model::Rackspace }.include? true
```
Comment (ekohl) [inline config/initializers/deprecations.rb]: Isn't this equivalent?
```suggestion
    next unless ComputeResource.all.unscoped.any? { |cr| cr.class == Foreman::Model::Rackspace }
```

**Review by ekohl — COMMENTED** [review]
Is the Foreman::Model::Rackspace correctly loaded in production where `foreman-rackspace` might not be installed.

File: config/initializers/deprecations.rb (hunk 1 above)
Comment (tbrisker) [reply config/initializers/deprecations.rb]: We can bypass the uninitialized constant issue by checking the `type` field, something like: 
```suggestion
    next unless ComputeResource.all.unscoped.where(type: 'Foreman::Model::Rackspace').any?
```

File: config/initializers/deprecations.rb [hunk 2]
```diff
@@ -1 +1,28 @@
 ActiveSupport::Deprecation.behavior = :silence if Rails.env.production?
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    next if Foreman.in_rake?
+    next unless Rails.env.production?
+    next unless ComputeResource.all.unscoped.where(type: 'Foreman::Model::Rackspace').any?
+    blueprint = NotificationBlueprint.find_by_name('feature_deprecation')
+    next unless blueprint
+    message = UINotifications::StringParser.new(blueprint.message, {feature: 'Rackspace Compute Resource', version: '2.0'}).to_s
```
Comment (tbrisker) [inline config/initializers/deprecations.rb]: the version should be 2.1 - deprecations state the version in which something will be removed.

**Review by tbrisker — APPROVED** [review]
Thanks @shiramax !

**Comment by tbrisker** [comment]
2.0 - ebc8ce47d

**Comment by tbrisker** [comment]
Please also add this to the deprecations section in https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/2.0/1.2_release_notes.md

**Pull request merged**